aspnetcore/mvc/controllers/bind-tryparse/7.0-samples/BindUsingTryParse/BindTryParseMVC/Models/DateRangeTP.cs

```diff
@@ -0,0 +1,33 @@
+namespace BindTryParseMVC.Models;
+
+// <snippet>
+public class DateRangeTP
+{
+    public DateOnly? From { get; }
+    public DateOnly? To { get; }
+
+    public DateRangeTP(string from, string to)
+    {
+        if (string.IsNullOrEmpty(from))
+            throw new ArgumentNullException(nameof(from));
+        if (string.IsNullOrEmpty(to))
+            throw new ArgumentNullException(nameof(to));
+
+        From = DateOnly.Parse(from);
+        To = DateOnly.Parse(to);
+    }
+
+    public static bool TryParse(string? value, out DateRangeTP? result)
+    {
+        if (string.IsNullOrEmpty(value) || value.Split('-').Length != 2)
+        {
+            result = default;
+            return false;
+        }
+
+        var range = value.Split(',');
+        result = new DateRangeTP(range[0], range[1]);
+        return true;
+    }
+}
```

Comment:
@fiyazbinhasan please review. I took out `IFormatProvider provider` because I don't think it's needed in the sample. If the code needs improvement, kindly create a PR>

Reply:
@fiyazbinhasan  it's live now, please review https://docs.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-7.0#bind-with-iparsablettryparse

Reply:
Also look at the following section https://docs.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-7.0#bind-with-tryparse